rdr_service/model/study_nph.py

```diff
@@ -12,12 +14,131 @@ class Participant(NphBase):
     id = Column("id", BigInteger, primary_key=True)
     created = Column(UTCDateTime)
     modified = Column(UTCDateTime)
-    ignore_flag = Column(TINYINT)
-    disable_flag = Column(TINYINT)
+    ignore_flag = Column(TINYINT, default=0)
+    disable_flag = Column(TINYINT, default=0)
     disable_reason = Column(String(1024))
     biobank_id = Column(BigInteger)
     research_id = Column(BigInteger)
 
 
 event.listen(Participant, "before_insert", model_insert_listener)
 event.listen(Participant, "before_update", model_update_listener)
+
+
+class StudyCategory(NphBase):
+    __tablename__ = "study_category"
+
+    id = Column("id", BigInteger, autoincrement=True, primary_key=True)
+    created = Column(UTCDateTime)
+    name = Column(String(128))
+    type_label = Column(String(128))
+    parent_id = Column(BigInteger, ForeignKey("study_category.id"))
+    parent = relation("StudyCategory", remote_side=[id])
+    children = relation("StudyCategory", remote_side=[parent_id], uselist=True)
+
+
+event.listen(StudyCategory, "before_insert", model_insert_listener)
```

Comment:
`model_insert_listener` will fail since StudyCategory doesn't have a "modified" field. The "created" field should be set to a default instead.

Reply:
@kenny-skaggs , I ran these unittests and they pass. Without any error

```python
class NphStudyCategoryTest(BaseTestCase):

    def setUp(self):
        super().setUp()
        self.nph_study_category_dao = NphStudyCategoryDao()

    def test_get_before_insert(self):
        self.assertIsNone(self.nph_study_category_dao.get(1))

    def _create_study_category(self, study_category_obj: Dict[str, Any]) -> StudyCategory:
        nph_study_category = StudyCategory(**study_category_obj)
        with FakeClock(TIME):
            return self.nph_study_category_dao.insert(nph_study_category)

    def test_insert_parent_study_category(self):
        parent_study_category = {
            "name": "Parent Study Category",
            "type_label": "PARENT",
            "parent_id": None
        }
        _parent_study_category = self._create_study_category(parent_study_category)
        expected_parent_study_category = {
            "id": 1,
            "created": TIME,
            "name": "Parent Study Category",
            "type_label": "PARENT",
            "parent_id": None
        }
        expected_parent_study_category_ = StudyCategory(**expected_parent_study_category)
        self.assertEqual(
            expected_parent_study_category_.asdict(),
            _parent_study_category.asdict()
        )

    def test_insert_child_study_category(self):
        parent_study_category_obj = {
            "name": "Parent Study Category",
            "type_label": "PARENT",
            "parent_id": None
        }
        parent_study_category = self._create_study_category(parent_study_category_obj)

        child_study_category_obj = {
            "name": "Child Study Category",
            "type_label": "CHILD",
            "parent_id": parent_study_category.id
        }
        child_study_category = self._create_study_category(child_study_category_obj)
        expected_child_study_category = {
            "id": 2,
            "created": TIME,
            "name": "Child Study Category",
            "type_label": "CHILD",
            "parent_id": parent_study_category.id
        }
        expected_child_study_category_ = StudyCategory(**expected_child_study_category)
        self.assertEqual(
            expected_child_study_category_.asdict(),
            child_study_category.asdict()
        )

    def tearDown(self):
        self.clear_table_after_test("nph.study_category")
```

Reply:
@jmtrujillo75, you're right. We're already doing the same thing here that we're doing with ApiUser (calling model_insert_listener without a modified field on the object). I'm not sure what I was conflating this with, but I was wrong to think it'd be an issue. Sorry about that!